### DISCUSSION TITLE
Allow direction of monorail trains to be reversed

### DISCUSSION THREAD

**Discussion by jcurbaniak** [OpenRCT2/OpenRCT2#22844] [Feature requests]
Unlike some other rides there does not seem to be any way to reverse the direction of monorail trains.  Or am I missing something?

**Comment by MASHzj** [comment]
Aren't the monorail trains symmetrical, so it would just look the same?

**Comment by 733737** [comment]
as MASHzj said, all of the different styles of monoral, be it retro, modern, or suspended are symmetrical front to back.

**Comment by jcurbaniak** [comment]
No, not necessarily.   When the track has to be raised to clear terrain or a ride, one side may be steeper than the other depending on the boundaries of the park.   

 

In these cases I prefer the train to come over the less steep side.   I have no control over the direction.

**Reply by MASHzj** [reply]
I don't understand what you mean. I don't see how reverse trains are related to steepness or the park boundary ...

**Reply by sjoerddebruin** [reply]
The "reversed trains" option doesn't change the direction the vehicle travels in.

**Comment by jcurbaniak** [comment]
I cannot explain it any better than I did.

**Comment by jcurbaniak** [comment]
It is not available for monorails.   Trains can travel either clockwise or counterclockwise.  

 

Otherwise what is the meaning of reverse?

**Reply by MASHzj** [reply]
Reverse trains means that the seats/guests are facing backwards instead of forwards on normal rides. The direction the train travels remains the same. Since monorail trains have both forward as backward seats in a symmetrical manner, reversing the train wouldn't make any difference, as others seem to confirm.

**Comment by jcurbaniak** [comment]
My monorail trains are traveling in a counter clockwise direction.   I prefer them to travel in a clockwise direction.    I cannot do that.

 

For a game with so many handy player options this seems an oversight to me.

**Reply by MASHzj** [reply]
I'm afraid you have to rebuild your track in that case. This is not unique to the monorail, no ride type can reverse the travel direction.

**Reply by spacek531** [reply]
If you want to reverse the trains you will have to use hacks. You will have to close the ride and change the operating mode to "Shuttle Mode", then force each train to reach the end of a track so that it changes direction. When you have done this with all the trains, the ride will be running in the opposite direction.

A plugin would be easier, but I believe the behavior flag that controls the direction of travel is not exposed to the plugin API.

**Comment by jcurbaniak** [comment]
Thanks for the suggestion.  But really this is too much hassle to be worth it.

 

For Shuttle Mode I would have to build a second station, then wait to the reversal, then remove the second station, then reconnect the line.

 

There should be in my view an easier way to reverse direction.

 

Now I grant this would not work for steam trains as it would not be possible to reposition the engine.

 

But seems like maybe it could work for monorail trains.

 

And rebuilding the entire line is just ridiculous.

 

Well it still remains a great game and I have spent many happy hours with it.
